src/ray/pubsub/publisher.cc

```diff
@@ -188,54 +191,53 @@ bool Subscriber::PublishIfPossible(bool force_noop) {
   return true;
 }
 
-bool Subscriber::CheckNoLeaks() const {
-  return !long_polling_connection_ && mailbox_.size() == 0;
+bool SubscriberState::CheckNoLeaks() const {
+  // If all message in the mailbox has been replied, consider there is no leak.
+  return !long_polling_connection_ && mailbox_.empty();
 }
 
-bool Subscriber::IsDisconnected() const {
-  return !long_polling_connection_ &&
-         get_time_ms_() - last_connection_update_time_ms_ >= connection_timeout_ms_;
+bool SubscriberState::ConnectionExists() const {
+  return long_polling_connection_ != nullptr;
 }
 
-bool Subscriber::IsActiveConnectionTimedOut() const {
-  return long_polling_connection_ &&
-         get_time_ms_() - last_connection_update_time_ms_ >= connection_timeout_ms_;
+bool SubscriberState::IsActive() const {
+  return get_time_ms_() - last_connection_update_time_ms_ < connection_timeout_ms_;
 }
 
 }  // namespace pub_internal
 
-void Publisher::ConnectToSubscriber(const SubscriberID &subscriber_id,
+void Publisher::ConnectToSubscriber(const rpc::PubsubLongPollingRequest &request,
```

Comment:
Is this change necessary? Are we planning to add more fields to the request message later?